Analysis/Tasks/PWGHF/D0D0barCorrelator.cxx

```diff
@@ -0,0 +1,850 @@
+        continue;
+      }
+      if (cutYCandMax >= 0. && std::abs(RecoDecay::Y(array{particle1.px(), particle1.py(), particle1.pz()}, RecoDecay::getMassPDG(particle1.pdgCode()))) > cutYCandMax) {
+        continue;
+      }
+      if (cutPtCandMin >= 0. && std::abs(particle1.pt()) < cutPtCandMin) {
+        continue;
+      }
+      registry.fill(HIST("hPtCandMCGen"), particle1.pt());
+      registry.fill(HIST("hEtaMCGen"), particle1.eta());
+      registry.fill(HIST("hPhiMCGen"), particle1.phi());
+      registry.fill(HIST("hYMCGen"), RecoDecay::Y(array{particle1.px(), particle1.py(), particle1.pz()}, RecoDecay::getMassPDG(particle1.pdgCode())));
+      counterD0D0bar++;
+
+      //D-Dbar correlation dedicated section
+      //if it's a D0 particle, search for D0bar and evaluate correlations
+      if (particle1.pdgCode() == 421) {                                  //just checking the particle PDG, not the decay channel (differently from Reco: you have a BR factor btw such levels!)
+        registry.fill(HIST("hcountD0triggersMCGen"), 0, particle1.pt()); //to count trigger D0 (for normalisation)
+        for (auto& particle2 : particlesMC) {
+          if (particle2.pdgCode() == -421) {
+            if (cutYCandMax >= 0. && std::abs(RecoDecay::Y(array{particle2.px(), particle2.py(), particle2.pz()}, RecoDecay::getMassPDG(particle2.pdgCode()))) > cutYCandMax) {
+              continue;
+            }
+            if (cutPtCandMin >= 0. && std::abs(particle2.pt()) < cutPtCandMin) {
+              continue;
+            }
+            entryD0D0barPair(getDeltaPhi(particle2.phi(), particle1.phi()),
+                             particle2.eta() - particle1.eta(),
+                             particle1.pt(),
+                             particle2.pt());
+            double etaCut = 0.;
+            double ptCut = 0.;
+            do { //fill pairs vs etaCut plot
+              ptCut = 0.;
+              etaCut += incrementEtaCut;
+              do { //fill pairs vs etaCut plot
+                if (std::abs(particle1.eta()) < etaCut && std::abs(particle2.eta()) < etaCut && particle1.pt() > ptCut && particle2.pt() > ptCut)
+                  registry.fill(HIST("hDDbarVsEtaCut"), etaCut - epsilon, ptCut + epsilon);
+                ptCut += incrementPtThreshold;
+              } while (ptCut < ptThresholdForMaxEtaCut - epsilon);
+            } while (etaCut < maxEtaCut - epsilon);
+          } // end D0bar check
+        }   //end inner loop
+      }     //end D0 check
+
+    } //end outer loop
+    registry.fill(HIST("hcountD0D0barPerEvent"), counterD0D0bar);
+  }
+};
+
+/// D0-D0bar correlation pair builder - LIKE SIGN - for real data and data-like analysis (i.e. reco-level w/o matching request via MC truth)
+/// NOTE: At the moment, both dPhi-symmetrical correlation pairs (part1-part2 and part2-part1) are filled, since we bin in pT and selecting as trigger the largest pT particle would bias the distributions w.r.t. the ULS case.
```

Comment:
This a super long comment. Please split into lines of 120 chars max.

Reply:
Done